packages/client/hmi-client/src/model-representation/petrinet/petrinet-service.ts

```diff
@@ -200,24 +207,33 @@ export const parseAMRPetriNet2IGraph = (
 		edges: []
 	};
 	const petrinet = amr.model as PetriNetModel;
+	const typing: TypingSemantics | undefined = amr.semantics?.typing;
 	// add each nodes in S
 	for (let i = 0; i < petrinet.states.length; i++) {
 		const aNode = petrinet.states[i];
+		const typeMap = typing?.type_map.find((map) => map.length === 2 && aNode.id === map[0]);
+		const strataType = typeMap?.[1] ?? '';
 		result.nodes.push({
 			id: aNode.id,
 			label: aNode.name,
 			x: 0,
 			y: 0,
 			height: config.S.height,
 			width: config.S.width,
-			data: { type: NodeType.State, uid: aNode.grounding.identifiers.ido },
+			data: {
+				type: NodeType.State,
+				uid: aNode.grounding?.identifiers?.ido ?? undefined,
+				strataType
+			},
 			nodes: []
 		});
 	}
 
 	// Add each node found in T
 	for (let i = 0; i < petrinet.transitions.length; i++) {
 		const aTransition = petrinet.transitions[i];
+		const typeMap = typing?.type_map.find((map) => map.length === 2 && aTransition.id === map[0]);
+		const strataType = typeMap?.[1] ?? '';
```

Comment:
Same as above, hard to to understand 
```
map.length === 2 && aNode.id === map[0]
```

Maybe consider a helper function and give it a meaningful name.

Reply:
I think a helper function for a one liner would be too verbose but I can definitely add an explanatory comment.